common/datasets/sms_wsj/returnn_datasets.py

```diff
@@ -0,0 +1,606 @@
+        :param str rasr_corpus: RASR corpus file for reading segment start and end times for padding
+        :param Optional[int] pad_label: target label assigned to padded areas
+        :param Optional[str] zip_cache: zip archive with SMS-WSJ data which can be cached, unzipped and used as data dir
+        """
+        data_types = {
+            "target_signals": {"dim": 2, "shape": (None, 2)},
+            "target_rasr": {
+                "sparse": True,
+                "dim": rasr_num_outputs,
+                "shape": (None, 2),
+            },
+        }
+        sms_wsj_base = SmsWsjBaseWithRasrClasses(
+            dataset_name=dataset_name,
+            json_path=json_path,
+            pre_batch_transform=self._pre_batch_transform,
+            scenario_map_args={"add_speech_reverberation_early": True},
+            rasr_classes_hdf=rasr_classes_hdf,
+            rasr_corpus=rasr_corpus,
+            rasr_segment_prefix=rasr_segment_prefix,
+            rasr_segment_postfix=rasr_segment_postfix,
+            pad_label=pad_label,
+            zip_cache=zip_cache,
+            data_types=data_types,
+        )
+        super(SmsWsjMixtureEarlyAlignmentDataset, self).__init__(
+            dataset_name,
+            json_path,
+            num_outputs=num_outputs,
+            zip_cache=zip_cache,
+            sms_wsj_base=sms_wsj_base,
+            **kwargs,
+        )
+        if num_outputs is not None:
+            self.num_outputs = num_outputs
+        else:
+            assert (
+                rasr_num_outputs is not None
+            ), "either num_outputs or rasr_num_outputs has to be given"
+            self.num_outputs["target_rasr"] = [
+                rasr_num_outputs,
+                1,
+            ]  # target alignments are sparse with the given dim
+
+
+class SmsWsjMixtureEarlyBpeDataset(SmsWsjMixtureEarlyDataset):
+    """
+    Dataset with audio mixture, target early signals and target BPE labels.
+    """
+
+    def __init__(
+        self,
+        dataset_name,
+        json_path,
+        bpe,
+        text_proc=None,
+        num_outputs=None,
+        zip_cache=None,
+        **kwargs,
+    ):
+        """
+        :param str dataset_name: "train_si284", "cv_dev93" or "test_eval92"
+        :param str json_path: path to SMS-WSJ json file
+        :param Dict[str] bpe: opts for :class:`BytePairEncoding`
+        :param Optional[Callable] text_proc: function to preprocess the transcriptions before applying BPE
+        :param Optional[Dict[str, List[int]]] num_outputs: num_outputs for RETURNN dataset
+        :param Optional[str] zip_cache: zip archive with SMS-WSJ data which can be cached, unzipped and used as data dir
+        """
+        sms_wsj_base = SmsWsjBase(
+            dataset_name=dataset_name,
+            json_path=json_path,
+            pre_batch_transform=self._pre_batch_transform,
+            scenario_map_args={"add_speech_reverberation_early": True},
+            zip_cache=zip_cache,
```

Comment:
There is a small error here. `SmsWsjBase` requires a `data_types` argument.
I think the definition should be similar to `SmsWsjMixtureEarlyAlignmentDataset`, so something like
```
data_types = {
            "target_signals": {"dim": 2, "shape": (None, 2)},
            "target_bpe": {
                "sparse": True,
                "dim": num_labels,
                "shape": (None, 2),
            },
        }
```

In this case, `num_labels` would be a new parameter.

Reply:
Thanks for spotting this!